traversal/planner/PlannerEdge.java

```diff
@@ -1287,7 +1287,7 @@ void updateObjective(GraphManager graphMgr) {
                         Set<TypeVertex> roleTypeVertices = iterate(this.roleTypes()).map(graphMgr.schema()::getType).toSet();
                         for (TypeVertex roleType : roleTypeVertices) {
                             assert roleType.isRoleType() && roleType.properLabel().scope().isPresent();
-                            double div = graphMgr.data().stats().thingVertexCount(Label.of(roleType.properLabel().scope().get()));
+                            double div = graphMgr.data().stats().thingVertexTransitiveCount(Label.of(roleType.properLabel().scope().get()));
```

Comment:
You posed this question to @dmitrii-ubskii which has not been answer, but you have approved the PR, @flyingsilverfin ? Does that mean we have verified that it's correct for me to approve the PR? From the sound of your question above, it can't be?

Reply:
I also need to review and approve the PR, ideally after I know the answer to the question above.

Reply:
When a role in question is inherited, the scope part of its label (say `base:player_1` with a bunch of `derived sub base`) refers to the supertype alone (`base` in this example). If we want to estimate how many roles of the given type will be instantiated for each relation, we have to include subtypes of the relation — the supertype may not even have any instances in the first place.
Since the scope of the role is not limited by the actual direct type of the relation involved, we can't just limit our statistics to the list of relation types in the vertex (`thingVertexSum(from.props().types())`), as that would lead to an overestimate of the cost of traversing the edge. OTOH, this version may lead to an underestimate if a subtype relation overrides the role type. Overall I expect role overrides to be rarer, so I went with this version.

Reply:
Illustration:
```graphql
base sub relation,     # base: 0 instances
    relates player;    # base:player: R instances

derived-1 sub base;  # D1 instances
derived-2 sub base;  # D2 instances
derived-3 sub base;  # D3 instances

derived-4 sub base,  # D4 instances
    relates cd-player as player;
```
We want, ideally, R/(D1+D2+D3) for the `base:player` term.
Before the change we had R/0.
The current implementation gives R/(D1+D2+D3+D4).
The alternate implementation would give R/D1.

Reply:
That makes sense! Thanks for the great explanation @dmitrii-ubskii !